app/controllers/tournaments_controller.rb

```diff
@@ -74,6 +76,24 @@ def destroy
 
   private
 
+  def organize_teams_in_groups(teams)
+    # each team gets put into a array of teams depending on the group specified in team[:group]
+    teams.group_by { |team| team['group'] }.values.map do |group|
+      group.map do |team|
+        find_or_create_team(team)
+      end
+    end
+  end
+
+  def find_or_create_team(team)
```

Comment:
[#find_or_create_by](https://devdocs.io/rails~5.2/activerecord/relation#method-i-find_or_create_by) just found this built-in method doing the exact same thing, please use that one instead

Reply:
Are you sure we can use that? (We are searching for id or creating for which we use the name.)

Reply:
If you look at all examples on that page, there are some where the searched attribute differs from the attributes used for creating the object.

Reply:
Reading through all examples is not really a strength of mine..

Reply:
You mean like this?

`Team.create_with(name: 'Düsemann').find_or_create_by(id: 42)`

Reply:
According to your tests this works, gonna change it :+1:

Reply:
As just discussed, I will not use the function as it fulfills another use case that we do not require.